### PR TITLE
macOS Clang builds

### PR DESCRIPTION
**Pull request by kmatt**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- Bug Fix
- Documentation

Changelog entry:

- Remove unsupported compiler flag on Clang / macOS (-Wno-ctad-maybe-unsupported)
- Update build docs for Clang / macOS

Addresses:

- #7995
- #7546
- #5765

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
This PR breaks clang builds: https://clickhouse-builds.s3.yandex.net/8148/54ddf4b0a03a12c6d2e6a5e5e78d88087ca18b1d/report.html

File: docs/en/development/build_osx.md [hunk 1]
```diff
@@ -33,11 +33,23 @@ For the latest stable version, switch to the `stable` branch.
 ```bash
 $ mkdir build
 $ cd build
-$ cmake .. -DCMAKE_CXX_COMPILER=`which g++-8` -DCMAKE_C_COMPILER=`which gcc-8`
+$ cmake .. -DCMAKE_CXX_COMPILER=`which g++` -DCMAKE_C_COMPILER=`which gcc`
```
Comment (kolen) [inline docs/en/development/build_osx.md]: Current version of this doc  in `master` has ``` -DCMAKE_CXX_COMPILER=`which clang++` -DCMAKE_C_COMPILER=`which clang` ``` here. So probably this line no longer requires any changes. 

On Mac OS, `gcc` binary is a shortcut to clang, but I don't know if it has any differences when invoked as `gcc` or `clang`. But error that had been fixed in this PR occurs both when using `gcc` and when using `clang`.

File: docs/en/development/build_osx.md [hunk 2]
```diff
@@ -33,11 +33,23 @@ For the latest stable version, switch to the `stable` branch.
...
 $ ninja
 $ cd ..
 ```
 
+## Build ClickHouse client only
+
+```bash
+$ mkdir build
+$ cd build
+$ cmake .. -DENABLE_CLICKHOUSE_ALL=0 -DENABLE_CLICKHOUSE_CLIENT=1 -DENABLE_MYSQL=0 -DCMAKE_CXX_COMPILER=`which g++` -DCMAKE_C_COMPILER=`which gcc`
```
Comment (kolen) [inline docs/en/development/build_osx.md]: According to current `master`, here should be ``` -DCMAKE_CXX_COMPILER=`which clang++` -DCMAKE_C_COMPILER=`which clang` ``` too.

Maybe doc changes are better to be separated to another pull request?

**Comment by zhang2014** [comment]
I have also investigated this problem,  and I'm not sure if it helps you(hopefully):

#7621
https://reviews.llvm.org/D56731
https://github.com/llvm/llvm-project/commit/73b51ae160af6b94d4468331ff3fb6855cff3b18 
this commit exists at  `master` `llvmorg-10-init`  `llvmorg-9.0.1-rc3` `llvmorg-9.0.1-rc2` `llvmorg-9.0.1-rc1` `llvmorg-9.0.0` `llvmorg-9.0.0-rc6` `llvmorg-9.0.0-rc5` `llvmorg-9.0.0-rc4` `llvmorg-9.0.0-rc3` `llvmorg-9.0.0-rc2` `llvmorg-9.0.0-rc1`

Apple clang corresponds to a possible version of llvm clang:
https://en.wikipedia.org/wiki/Xcode#Latest_versions

**Comment by alexey-milovidov** [comment]
Unknown flag: fixed in https://github.com/ClickHouse/ClickHouse/commit/cea2903e72d4378b382f9b8ae616c01f512483dc#diff-eabb2b894359aa83d33168e3117959dcR55

Build instruction: obsolete.
